In [9]:
import torch
import torch.nn as nn
import torch.nn.functional as F
import torchvision
import torchaudio
import numpy as np
import scipy
import sklearn
import pandas as pd
import PIL
import matplotlib
import tokenizers
import datasets
import transformers
import matplotlib.pyplot as plt
from collections import Counter
import datetime

# Data exploration

Here we explore each feature column and comment on how we are going to preprocess each one. We have a total of 140700 samples. The last feature is the label depression which we will train on.

## Id column



In [17]:
# Load the dataset
data = pd.read_csv("./data/train.csv")

# Print examples of the 'id' column
print("Examples of 'id' column:")
print(data["id"].head())

print("")

print("We have " ,data["id"].count()," amount of samples")

Examples of 'id' column:
0    0
1    1
2    2
3    3
4    4
Name: id, dtype: int64

We have  140700  amount of samples


We will drop the id column since there is no correlation between this and the label.

## Name column

In [19]:
# Print examples of the 'Name' column
print("Examples of 'Name' column:")
print(data["Name"].head())

print("")

print("Number of unique names:", data["Name"].nunique())

Examples of 'Name' column:
0    Aaradhya
1       Vivan
2      Yuvraj
3      Yuvraj
4        Rhea
Name: Name, dtype: object

Number of unique names: 422
Number of names:  140700


We thought about dropping the name column since we thought that this may not have any correlation to the label. But we realised that there might be a correlation, so we decided to keep it. The correlation being that for example having a "unattractive" name can affect your life in a bad way. 


For pre-processing this we need to one-hot code each name so we have 422 unique names so we will have them from 0 to 421. Here it is important to make sure that when we pre-process the test data the same names get the same number.

# Pre-processing

Dropping the Id column

In [ ]:
# Drop the 'id' column
data = data.drop(columns=["id"])

# Display the remaining features
print("Features left after dropping 'id':")
print(data.columns.tolist())